### PR TITLE
feat: new api for get shorten url details

### PR DESCRIPTION
**Pull request by sahsisunny**
## Description

This pull request introduces a new API feature that enhances our URL shortening service by providing two key functionalities:

1. **Get Shortened URL Details API:** This API allows users to retrieve detailed information about a specific shortened URL. It provides insights into the original URL, creation date, click statistics, and any custom alias associated with the shortened URL.

2. **Get All URLs Created by User API:** This API enables users to obtain a list of all URLs they have created using our service, based on their user ID.

### CODE REVIEW

**Comment by vinit717** [comment]
Please provide ss of tests stats

File: routes/users.go [hunk 1]
```diff
@@ -8,8 +8,11 @@ import (
 )
 
 func UserRoutes(rg *gin.RouterGroup, db *bun.DB) {
+
     users := rg.Group("/users")
+    user := rg.Group("/user")
     users.Use(middleware.AuthMiddleware())
```
Comment (vinit717) [inline routes/users.go:14]: why there is 2 route /users and /user

Comment (sahsisunny) [reply routes/users.go:14]: See API endpoints i shared in the discord channel

**Comment by sahsisunny** [comment]
> Please provide ss of tests stats

Can you show me a screenshot of the test stats in the tiny site backend PR of a tiny site or in your PR which recently merged?

**Review by iamitprakash — APPROVED** [review]

**Review by shubhamsigdar1 — APPROVED** [review]

**Pull request merged**